src/qt/optionsdialog.cpp

```diff
@@ -112,6 +119,52 @@ OptionsDialog::OptionsDialog(QWidget *parent, bool enableWallet) :
 
     /* setup/change UI elements when proxy IP is invalid/valid */
     connect(this, SIGNAL(proxyIpChecks(QValidatedLineEdit *, int)), this, SLOT(doProxyIpChecks(QValidatedLineEdit *, int)));
+
+    uint64_t max,ave;
+    sendShaper.get(&max,&ave);
+    bool enabled = (ave != LONG_MAX);
+    ui->sendShapingEnable->setChecked(enabled);
+    ui->sendBurstSlider->setRange(0,1000);  // The slider is just for convenience so setting their ranges to what is commonly chosen
+    ui->sendAveSlider->setRange(0,1000);
+    ui->recvBurstSlider->setRange(0,1000);
+    ui->recvAveSlider->setRange(0,1000);
+
+    ui->sendBurstEdit->setValidator(&burstValidator);
+    ui->recvBurstEdit->setValidator(&burstValidator);
+    ui->sendAveEdit->setValidator(&sendAveValidator);
+    ui->recvAveEdit->setValidator(&recvAveValidator);
+   
+    connect(ui->sendBurstSlider, SIGNAL(valueChanged(int)), this, SLOT(shapingSliderChanged()));
+    connect(ui->sendAveSlider, SIGNAL(valueChanged(int)), this, SLOT(shapingSliderChanged()));
+    connect(ui->recvBurstSlider, SIGNAL(valueChanged(int)), this, SLOT(shapingSliderChanged()));
+    connect(ui->recvAveSlider, SIGNAL(valueChanged(int)), this, SLOT(shapingSliderChanged()));
+
+    connect(ui->recvAveEdit, SIGNAL(editingFinished()), this, SLOT(shapingAveEditFinished()));
+    connect(ui->sendAveEdit, SIGNAL(editingFinished()), this, SLOT(shapingAveEditFinished()));
+    connect(ui->recvBurstEdit, SIGNAL(editingFinished()), this, SLOT(shapingMaxEditFinished()));
+    connect(ui->sendBurstEdit, SIGNAL(editingFinished()), this, SLOT(shapingMaxEditFinished()));
+   
+    if (enabled)
+      {
```

Comment:
Again, please fix your editor so it doesn't indent every line a random amount. Thanks!